src/clj/game/cards/identities.clj

```diff
@@ -2106,17 +2108,21 @@
             {:event :runner-turn-ends
              :req (req (and (not (:disabled card))
                             (not (is-disabled? state side card))
-                            (has-most-faction? state :corp "Haas-Bioroid")
-                            (pos? (count (:discard corp)))))
-             :prompt "Choose a card in Archives to shuffle into R&D"
-             :choices {:card #(and (corp? %)
-                                   (in-discard? %))}
-             :player :corp
-             :show-discard true
-             :msg (msg "shuffle " (if (:seen target) (:title target) "a card")
-                       " into R&D")
-             :effect (effect (move :corp target :deck)
-                             (shuffle! :corp :deck))}]})
+                            (has-most-faction? state :corp "Haas-Bioroid")))
+             :async true
+             :effect (req (if (empty? (:discard corp))
+                            (do (shuffle-cards-into-deck! state :corp card :card [])
```

Comment:
`:card` is the target side having their deck shuffled?

Reply:
whoops